icu4c/source/i18n/unicode/tmutfmt.h

```diff
@@ -141,7 +141,15 @@ class U_I18N_API TimeUnitFormat: public MeasureFormat {
      * @return      true if the given Format objects are not semantically equal.
      * @deprecated ICU 53
      */
-    UBool operator!=(const Format& other) const;
+    bool operator!=(const Format& other) const;
+
+#if defined(__cpp_impl_three_way_comparison) && \
+       __cpp_impl_three_way_comparison >= 201711
+    // Manually resolve C++20 reversed argument order ambiguity.
+    inline bool operator!=(const TimeUnitFormat& other) const {
+        return operator!=(static_cast<const Format&>(other));
+    }
+#endif
```

Comment:
Can you please explain the problem? Why is there ambiguity here but not for other `operator!=()` without this extra code?

Could we avoid it by conditionally (>= C++20) removing the `operator!=()` altogether?

If we need a new public API function like this, then we should add API comments, so that it does not show up in the docs as untagged / with unknown API status.

Reply:
The compiler error message with C++20 without this workaround is:

```
icu4c/source/test/intltest/tufmtts.cpp:261:59: error: use of overloaded operator '!=' is ambiguous (with operand types 'icu_70::TimeUnitFormat' and 'icu_70::TimeUnitFormat')
    assertTrue("TimeUnitFormat: en and fr diff", (*tmf_en != tmf_fr));
                                                  ~~~~~~~ ^  ~~~~~~
icu4c/source/i18n/unicode/tmutfmt.h:240:17: note: candidate function
TimeUnitFormat::operator!=(const Format& other) const  {
                ^
icu4c/source/i18n/unicode/measfmt.h:151:18: note: candidate function
    virtual bool operator==(const Format &other) const;
                 ^
icu4c/source/i18n/unicode/measfmt.h:151:18: note: candidate function (with reversed parameter order)
```

According to C++20 resolution rules, the code is ambiguous becase `==` and `!=` have different left hand side data types (`MeasureFormat` vs. `TimeUnitFormat`).

Deleting the unnecessary and deprecated `TimeUnitFormat::operator!=()` would resolve the ambiguity and I hope to be able to do that before the RC, but that'd definitely be a public API change by anyone's definition, which is why I propose to start out with this workaround as a first step just to get this code to compile at all and so be able to set up C++20 continuous builds while working on getting rid of the workarounds.